sympy/sets/tests/test_fancysets.py

```diff
@@ -295,18 +295,162 @@ def test_Range_set():
+            rabc = r.subs({a: av, c: cv}).subs(b, oo)
+            if (rinf is S.NaN):
+                assert not rabc
+                continue
+            assert rinf == rabc.inf
+            rinf = r.inf.subs({a: av, c: cv}).subs(b, -oo)
+            rabc = r.subs({a: av, c: cv}).subs(b, -oo)
+            if (rinf is S.NaN):
+                assert not rabc
+                continue
+            assert rinf == rabc.inf
+
+    for av in avs:
+        for bv in bvs:
+            for cv in cvs:
+                rsup = r.sup.subs({a: av, b: bv, c: cv})
+                rabc = r.subs({a: av, b: bv, c: cv})
+                if (rsup is S.NaN):
+                    assert not rabc
+                    continue
+                assert rsup == rabc.sup
+    for av in avs:
+        for cv in cvs:
+            rsup = r.sup.subs({a: av, c: cv}).subs(b, oo)
+            rabc = r.subs({a: av, c: cv}).subs(b, oo)
+            if (rsup is S.NaN):
+                assert not rabc
+                continue
+            assert rsup == rabc.sup
+            rsup = r.sup.subs({a: av, c: cv}).subs(b, -oo)
+            rabc = r.subs({a: av, c: cv}).subs(b, -oo)
+            if (rsup is S.NaN):
+                assert not rabc
+                continue
+            assert rsup == rabc.sup
+
+    avs, bvs = ([0, 20, -30], )*2
+    cvs = [1, -2]
+    idxs = [0, 3, -10, oo, -1]
+
+    def check_gtm(rgtm, rabc, idx):
+        if not rabc:
+            raises(IndexError, lambda : rabc.__getitem__(idx))
+        elif rgtm in (S.Infinity, S.NegativeInfinity) and idx not in (oo, -1, 0):
+            raises(ValueError, lambda : rabc.__getitem__(idx))
+        else:
+            assert rgtm == rabc.__getitem__(idx)
+
+    for av in avs:
+        for bv in bvs:
+            for cv in cvs:
+                for idx in idxs:
+                    rgtm = r.__getitem__(idx).subs({a: av, b: bv, c: cv})
+                    rabc = r.subs({a: av, b: bv, c: cv})
+                    check_gtm(rgtm, rabc, idx)
+    for av in avs:
+        for cv in cvs:
+            for idx in idxs:
+                rgtm = r.__getitem__(idx).subs({a: av, c: cv}).subs(b, oo)
+                rabc = r.subs({a: av, c: cv}).subs(b, oo)
+                check_gtm(rgtm, rabc, idx)
+                rgtm = r.__getitem__(idx).subs({a: av, c: cv}).subs(b, -oo)
+                rabc = r.subs({a: av, c: cv}).subs(b, -oo)
+                check_gtm(rgtm, rabc, idx)
+
+    assert Range(range(10)) == Range(10)
+    assert Range(range(1, 10)) == Range(1, 10)
+    assert Range(range(1, 10, 2)) == Range(1, 10, 2)
     if PY3:
-        builtin_range = range
-    else:
-        builtin_range = xrange
-
-    assert Range(builtin_range(10)) == Range(10)
-    assert Range(builtin_range(1, 10)) == Range(1, 10)
-    assert Range(builtin_range(1, 10, 2)) == Range(1, 10, 2)
-    if PY3:
-        assert Range(builtin_range(1000000000000)) == \
+        assert Range(range(1000000000000)) == \
```

Comment:
If range is imported from sympy.core.compatilbity then it will be xrange on PY2 so there is no need for the PY3 check above.